Fix a couple corner cases related to type parsing.  This was causing
parseTypesNow() to crash on some BG/Q binaries (miniFE) produced by
the XL compiler.


diff -Naurb dyninst.orig/symtabAPI/src/dwarfWalker.C dyninst/symtabAPI/src/dwarfWalker.C
--- dyninst.orig/symtabAPI/src/dwarfWalker.C	2014-03-12 18:20:10.000000000 -0500
+++ dyninst/symtabAPI/src/dwarfWalker.C	2014-04-15 15:15:05.672715518 -0500
@@ -756,6 +756,7 @@
 
    Type *type = NULL;
    if (!findType(type, false)) return false;
+   assert(type);
    
    Dwarf_Unsigned variableLineNo;
    bool hasLineNumber = false;
@@ -1429,6 +1430,7 @@
 }
 
 bool DwarfWalker::findType(Type *&type, bool defaultToVoid) {
+  // Do *not* return true unless type is actually usable.
    int status;
 
    /* Acquire the parameter's type. */
@@ -1439,8 +1441,9 @@
    if (status == DW_DLV_NO_ENTRY) {
       if (defaultToVoid) {
          type = tc()->findType("void");
+	 return (type != NULL);
       }
-      return true;
+      return false;
    }
 
    Dwarf_Off typeOffset;
@@ -1471,8 +1474,15 @@
       Dwarf_Unsigned fileNameDeclVal;
       DWARF_FAIL_RET(dwarf_formudata(fileDeclAttribute, &fileNameDeclVal, NULL));
       dwarf_dealloc( dbg(), fileDeclAttribute, DW_DLA_ATTR );			
+      if(fileNameDeclVal) 
+      {
       fileName = srcFiles()[fileNameDeclVal-1];
    }
+      else
+      {
+	fileName = "";
+      }
+   }
    else {
       return true;
    }
